From patchwork Wed Feb 13 08:38:58 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 8bit
X-Patchwork-Submitter: "tip-bot for Paul E. McKenney" <tipbot@zytor.com>
X-Patchwork-Id: 10809333
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 1A2B11390
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 13 Feb 2019 08:39:08 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 06A352BF22
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 13 Feb 2019 08:39:08 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id EE7822BF60; Wed, 13 Feb 2019 08:39:07 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-7.9 required=2.0 tests=BAYES_00,MAILING_LIST_MULTI,
	RCVD_IN_DNSWL_HI autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 9A55B2BF22
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 13 Feb 2019 08:39:07 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S2390730AbfBMIjG (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Wed, 13 Feb 2019 03:39:06 -0500
Received: from terminus.zytor.com ([198.137.202.136]:51257 "EHLO
        terminus.zytor.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1733035AbfBMIjG (ORCPT <rfc822;kvm@vger.kernel.org>);
        Wed, 13 Feb 2019 03:39:06 -0500
Received: from terminus.zytor.com (localhost [127.0.0.1])
        by terminus.zytor.com (8.15.2/8.15.2) with ESMTPS id x1D8cwpt190697
        (version=TLSv1.3 cipher=TLS_AES_256_GCM_SHA384 bits=256 verify=NO);
        Wed, 13 Feb 2019 00:38:58 -0800
Received: (from tipbot@localhost)
        by terminus.zytor.com (8.15.2/8.15.2/Submit) id x1D8cwXj190694;
        Wed, 13 Feb 2019 00:38:58 -0800
Date: Wed, 13 Feb 2019 00:38:58 -0800
X-Authentication-Warning: terminus.zytor.com: tipbot set sender to
 tipbot@zytor.com using -f
From: "tip-bot for Paul E. McKenney" <tipbot@zytor.com>
Message-ID: <tip-6706dae90d85d394d0134d0ec516c25560b9ce77@git.kernel.org>
Cc: kvm@vger.kernel.org, hpa@zytor.com, tglx@linutronix.de,
        paulmck@linux.ibm.com, mingo@kernel.org, rkrcmar@redhat.com,
        pbonzini@redhat.com
Reply-To: hpa@zytor.com, kvm@vger.kernel.org, paulmck@linux.ibm.com,
          linux-kernel@vger.kernel.org, tglx@linutronix.de,
          rkrcmar@redhat.com, mingo@kernel.org, pbonzini@redhat.com
To: linux-tip-commits@vger.kernel.org
Subject: [tip:core/rcu] virt/kvm: Replace spin_is_locked() with lockdep
Git-Commit-ID: 6706dae90d85d394d0134d0ec516c25560b9ce77
X-Mailer: tip-git-log-daemon
Robot-ID: <tip-bot.git.kernel.org>
Robot-Unsubscribe: Contact <mailto:hpa@kernel.org> to get blacklisted from
 these emails
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Disposition: inline
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

Commit-ID:  6706dae90d85d394d0134d0ec516c25560b9ce77
Gitweb:     https://git.kernel.org/tip/6706dae90d85d394d0134d0ec516c25560b9ce77
Author:     Paul E. McKenney <paulmck@linux.ibm.com>
AuthorDate: Tue, 8 Jan 2019 10:39:49 -0800
Committer:  Paul E. McKenney <paulmck@linux.ibm.com>
CommitDate: Fri, 25 Jan 2019 15:36:05 -0800

virt/kvm: Replace spin_is_locked() with lockdep

lockdep_assert_held() is better suited to checking locking requirements,
since it only checks if the current thread holds the lock regardless of
whether someone else does. This is also a step towards possibly removing
spin_is_locked().

Signed-off-by: Paul E. McKenney <paulmck@linux.ibm.com>
Cc: Paolo Bonzini <pbonzini@redhat.com>
Cc: "Radim Krčmář" <rkrcmar@redhat.com>
Cc: <kvm@vger.kernel.org>
Acked-by: Paolo Bonzini <pbonzini@redhat.com>
---
 virt/kvm/kvm_main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/virt/kvm/kvm_main.c b/virt/kvm/kvm_main.c
index 5ecea812cb6a..28de87ff9f51 100644
--- a/virt/kvm/kvm_main.c
+++ b/virt/kvm/kvm_main.c
@@ -4083,7 +4083,7 @@ static int kvm_suspend(void)
 static void kvm_resume(void)
 {
 	if (kvm_usage_count) {
-		WARN_ON(raw_spin_is_locked(&kvm_count_lock));
+		lockdep_assert_held(&kvm_count_lock);
 		hardware_enable_nolock(NULL);
 	}
 }
